device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java

```diff
@@ -160,8 +163,9 @@ else if (attr.startsWith(SHARED_ACCESS_TOKEN_ATTRIBUTE))
         }
 
         initIotHubClient(hostname, deviceId, sharedAccessKey, sharedAccessToken, protocol);
-        this.logger = new CustomLogger(this.getClass());
-        logger.LogInfo("DeviceClient object is created successfully, method name is %s ", logger.getMethodName());
+        
+        logger.info("DeviceClient object is created successfully, method name is {} ", 
+        		Thread.currentThread().getStackTrace()[2].getMethodName());
```

Comment:
This was just copied from the original code. BTW. It's wrong, too, because the wrong method name gets returned. See Issue #58 . It should completely removed or exchanged by putting the method name directly into the log string, since calling the stack trace is very very slow!